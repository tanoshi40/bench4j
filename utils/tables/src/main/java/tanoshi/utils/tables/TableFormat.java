package tanoshi.utils.tables;

import java.util.Arrays;
import java.util.stream.Collectors;

public enum TableFormat {
    MINIMAL {
        @Override
        public String buildConnectorTemplate(int[] columnWidths) {
            StringBuilder builder = new StringBuilder();

            char[] preChars = new char[columnWidths[0] + 1];
            Arrays.fill(preChars, 'h');
            builder.append(String.valueOf(preChars));

            for (int i = 1; i < columnWidths.length - 1; i++) {
                builder.append('c');

                int columnWidth = columnWidths[i];

                char[] chars = new char[columnWidth + 2];
                Arrays.fill(chars, 'h');
                builder.append(String.valueOf(chars));
            }

            builder.append('c');
            preChars = new char[columnWidths[columnWidths.length - 1] + 1];
            Arrays.fill(preChars, 'h');
            builder.append(String.valueOf(preChars));

            return builder.toString();
        }
    },
    DEFAULT {
        @Override
        public String buildConnectorTemplate(int[] columnWidths) {
            return TableFormat.buildStandardConnectorTemplate(columnWidths);
        }
    },
    LARGE {
        @Override
        public String buildConnectorTemplate(int[] columnWidths) {
            return TableFormat.buildStandardConnectorTemplate(columnWidths);
        }
    };


    public abstract String buildConnectorTemplate(int[] columnWidths);

    @SuppressWarnings("SpellCheckingInspection")
    private static String buildStandardConnectorTemplate(int[] columnWidths) {
        // "lhhhshhhshhhshr"
        StringBuilder builder = new StringBuilder();

        builder.append('l');

        builder.append(Arrays.stream(columnWidths).mapToObj(width -> {
            char[] chars = new char[width + 2];
            Arrays.fill(chars, 'h');
            return String.valueOf(chars);
        }).collect(Collectors.joining("c")));

        builder.append('r');
        return builder.toString();
    }
}
