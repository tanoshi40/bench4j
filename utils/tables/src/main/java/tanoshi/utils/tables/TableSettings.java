package tanoshi.utils.tables;

import tanoshi.utils.colors.ConsoleTextColor;

public class TableSettings {

    private TableFormat format = TableFormat.DEFAULT;
    private TableStyle style = TableStyle.DEFAULT;
    private ConsoleTextColor contentColor = null;

    public TableFormat getFormat() {
        return format;
    }

    public void setFormat(TableFormat tableFormat) {
        this.format = tableFormat;
    }

    public ConsoleTextColor getContentColor() {
        return contentColor;
    }

    public void setContentColor(ConsoleTextColor contentColor) {
        this.contentColor = contentColor;
    }

    public TableStyle getStyle() {
        return style;
    }

    public void setStyle(TableStyle style) {
        this.style = style;
    }

    public String colorizedContent(String content) {
        return contentColor == null ? content : contentColor.colorizeText(content);
    }
}