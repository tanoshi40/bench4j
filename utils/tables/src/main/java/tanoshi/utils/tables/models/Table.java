package tanoshi.utils.tables.models;

public class Table {
    private final String[] headers;
    private final String[][] content;
    private final String title;

    public Table(String[] headers, String[][] content, String title) {
        this.headers = headers;
        this.content = content;
        this.title = title;
    }
    public Table(String[] headers, String[][] content) {
        this(headers, content, null);
    }

    public String[] getHeaders() {
        return headers;
    }

    public String[][] getContent() {
        return content;
    }

    public String getTitle() {
        return title;
    }
}
