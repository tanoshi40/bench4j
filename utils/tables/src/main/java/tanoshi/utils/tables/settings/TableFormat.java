package tanoshi.utils.tables.settings;

import tanoshi.utils.colors.ConsoleTextColor;

import java.util.Arrays;
import java.util.stream.Collectors;

public enum TableFormat {
    MINIMAL {
        @Override
        public String buildConnectorTemplate(int[] columnWidths) {
            StringBuilder builder = new StringBuilder();

            char[] preChars = new char[columnWidths[0] + 1];
            Arrays.fill(preChars, 'h');
            builder.append(String.valueOf(preChars));

            for (int i = 1; i < columnWidths.length - 1; i++) {
                builder.append('c');

                int columnWidth = columnWidths[i];

                char[] chars = new char[columnWidth + 2];
                Arrays.fill(chars, 'h');
                builder.append(String.valueOf(chars));
            }

            builder.append('c');
            preChars = new char[columnWidths[columnWidths.length - 1] + 1];
            Arrays.fill(preChars, 'h');
            builder.append(String.valueOf(preChars));

            return colorize(builder.toString(),getLineColor());
        }
    },
    DEFAULT {
        @Override
        public String buildConnectorTemplate(int[] columnWidths) {
            return colorize(buildStandardConnectorTemplate(columnWidths), getLineColor());
        }
    },
    LARGE {
        @Override
        public String buildConnectorTemplate(int[] columnWidths) {
            return colorize(buildStandardConnectorTemplate(columnWidths), getLineColor());
        }
    };


    private ConsoleTextColor lineColor = null;

    public abstract String buildConnectorTemplate(int[] columnWidths);

    public void setLineColor(ConsoleTextColor lineColor) {
        this.lineColor = lineColor;
    }

    ConsoleTextColor getLineColor() {
        return lineColor;
    }

    @SuppressWarnings("SpellCheckingInspection")
    private static String buildStandardConnectorTemplate(int[] columnWidths) {
        // "lhhhshhhshhhshr"
        StringBuilder builder = new StringBuilder();

        builder.append('l');

        builder.append(Arrays.stream(columnWidths).mapToObj(width -> {
            char[] chars = new char[width + 2];
            Arrays.fill(chars, 'h');
            return String.valueOf(chars);
        }).collect(Collectors.joining("c")));

        builder.append('r');
        return builder.toString();
    }

    private static String colorize(String content, ConsoleTextColor color) {
        if (color == null) {
            return content;
        }
        return color.colorizeText(content);
    }
}
