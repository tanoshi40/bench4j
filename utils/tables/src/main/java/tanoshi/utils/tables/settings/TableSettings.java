package tanoshi.utils.tables.settings;

import tanoshi.utils.colors.ConsoleTextColor;

public class TableSettings {

    private TableFormat format = TableFormat.DEFAULT;
    private TableStyle style = TableStyle.DEFAULT;
    private ConsoleTextColor contentColor = null;
    private ConsoleTextColor lineColor = null;

    public TableFormat getFormat() {
        return format;
    }

    public TableSettings withFormat(TableFormat tableFormat) {
        this.format = tableFormat;
        return this;
    }

    public ConsoleTextColor getContentColor() {
        return contentColor;
    }

    public TableSettings withContentColor(ConsoleTextColor contentColor) {
        this.contentColor = contentColor;
        return this;
    }

    public TableSettings withLineColor(ConsoleTextColor lineColor) {
        this.lineColor = lineColor;
        return this;
    }

    public ConsoleTextColor getLineColor() {
        return lineColor;
    }

    public TableStyle getStyle() {
        return style;
    }

    public TableSettings withStyle(TableStyle style) {
        this.style = style;
        return this;
    }

    public String colorizedContent(String content) {
        return contentColor == null ? content : contentColor.colorizeText(content);
    }

    public String colorizedLineElement(String lineContent) {
        return lineColor == null ? lineContent : lineColor.colorizeText(lineContent);
    }
}