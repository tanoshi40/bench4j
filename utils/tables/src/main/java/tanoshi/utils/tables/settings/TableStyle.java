package tanoshi.utils.tables.settings;

public enum TableStyle {
    DEFAULT,
    RETRO
}
