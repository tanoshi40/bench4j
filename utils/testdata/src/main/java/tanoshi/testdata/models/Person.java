package tanoshi.testdata.models;

public class Person {

    private final String name;
    private final int age;
    private final int id;

    public Person(String name, int age, int id) {
        this.name = name;
        this.age = age;
        this.id = id;
    }
    public Person(Person person) {
        this(person.getName(), person.getAge(), person.getId());
    }

    public Person copy() {
        return new Person(this);
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public int getId() {
        return id;
    }

    @Override
    public String toString() {
        return "Person{" +
                "name='" + name + '\'' +
                ", age=" + age +
                ", id=" + id +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Person person)) return false;

        if (getAge() != person.getAge()) return false;
        if (getId() != person.getId()) return false;
        return getName() != null ? getName().equals(person.getName()) : person.getName() == null;
    }

    @Override
    public int hashCode() {
        int result = getName() != null ? getName().hashCode() : 0;
        result = 31 * result + getAge();
        result = 31 * result + getId();
        return result;
    }
}
