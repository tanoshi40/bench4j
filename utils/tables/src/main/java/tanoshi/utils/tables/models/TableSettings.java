package tanoshi.utils.tables.models;

import tanoshi.utils.colors.ConsoleTextColor;
import tanoshi.utils.tables.TableFormat;
import tanoshi.utils.tables.TableStyle;

public class TableSettings {

    private TableFormat format = TableFormat.DEFAULT;
    private TableStyle style = TableStyle.DEFAULT;
    private ConsoleTextColor contentColor = null;

    public TableFormat getFormat() {
        return format;
    }

    public TableSettings withFormat(TableFormat tableFormat) {
        this.format = tableFormat;
        return this;
    }

    public ConsoleTextColor getContentColor() {
        return contentColor;
    }

    public TableSettings withContentColor(ConsoleTextColor contentColor) {
        this.contentColor = contentColor;
        return this;
    }

    public TableStyle getStyle() {
        return style;
    }

    public TableSettings withStyle(TableStyle style) {
        this.style = style;
        return this;
    }

    public String colorizedContent(String content) {
        return contentColor == null ? content : contentColor.colorizeText(content);
    }
}