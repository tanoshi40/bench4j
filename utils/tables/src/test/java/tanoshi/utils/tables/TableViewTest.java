package tanoshi.utils.tables;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.EnumSource;
import tanoshi.testdata.models.Person;
import tanoshi.testdata.provider.PersonProvider;
import tanoshi.utils.tables.models.Table;
import tanoshi.utils.tables.settings.TableFormat;
import tanoshi.utils.tables.settings.TableSettings;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertNotNull;

@ExtendWith({SnapshotExtension.class})
class TableViewTest {

    private Expect expect;
    
    @ParameterizedTest
    @EnumSource(value = TableFormat.class)
    public void viewSimple(TableFormat format) {
        // Arrange
        Table table = buildTable(10, null);

        TableSettings settings = new TableSettings().withFormat(format);

        // Act
        TableView builder = TableView.build(table, settings);
        String tableString = builder.toView();

        // Assert
        assertNotNull(tableString, "table should not be null");

        expect.scenario(format.name()).toMatchSnapshot(tableString);
    }

    @ParameterizedTest
    @EnumSource(value = TableFormat.class)
    public void viewTitleSimple(TableFormat format) {
        // Arrange
        Table table = buildTable(10, "title");

        TableSettings settings = new TableSettings().withFormat(format);

        // Act
        TableView builder = TableView.build(table, settings);
        String tableString = builder.toView();

        // Assert
        assertNotNull(tableString, "table should not be null");

        expect.scenario(format.name()).toMatchSnapshot(tableString);
    }

    @ParameterizedTest
    @EnumSource(value = TableFormat.class)
    public void viewComplex(TableFormat format) {
        // Arrange
        String[] headers = getHeaders();
        Table[] tables = buildTables(false);

        TableSettings settings = new TableSettings().withFormat(format);

        // Act
        TableView builder = TableView.build(tables, headers, "complex table", settings);
        String tableString = builder.toView();

        // Assert
        assertNotNull(tableString, "table should not be null");

        expect.scenario(format.name()).toMatchSnapshot(tableString);
    }

    @ParameterizedTest
    @EnumSource(value = TableFormat.class)
    public void viewTitleComplex(TableFormat format) {
        // Arrange
        String[] headers = getHeaders();
        Table[] tables = buildTables(true);

        TableSettings settings = new TableSettings().withFormat(format);

        // Act
        TableView builder = TableView.build(tables, headers, "complex table", settings);
        String tableString = builder.toView();

        // Assert
        assertNotNull(tableString, "table should not be null");

        expect.scenario(format.name()).toMatchSnapshot(tableString);
    }

    private static String[] getHeaders() {
        return new String[]{"id", "name", "age"};
    }

    private static Table[] buildTables(boolean withTitles) {
        return new Table[]{
                buildTable(10, withTitles ? "table 1" : null),
                buildTable(5, withTitles ? "table 2" : null),
                buildTable(7, withTitles ? "table 3" : null)
        };
    }

    private static Table buildTable(int size, String title) {
        return Table.builder(getHeaders(), getContent(size))
                .withTitle(title).build();
    }

    private static String[][] getContent(int size) {
        String[][] content = new String[size][3];
        List<Person> persons = PersonProvider.getPersons(size);
        for (int i = 0; i < persons.size(); i++) {
            Person person = persons.get(i);
            content[i][0] = String.valueOf(person.getId());
            content[i][1] = person.getName();
            content[i][2] = String.valueOf(person.getAge());
        }
        return content;
    }
}