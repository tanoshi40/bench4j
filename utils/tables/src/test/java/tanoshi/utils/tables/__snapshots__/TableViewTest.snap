tanoshi.utils.tables.TableViewTest.viewComplex[DEFAULT]=[
┌───────────────────┐
│ complex table     │
├────┬────────┬─────┤
│ id │ name   │ age │
├────┼────────┼─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
│ 6  │ rho    │ 38  │
│ 7  │ omega  │ 86  │
│ 8  │ sigma  │ 64  │
│ 9  │ lambda │ 12  │
│ 10 │ beta   │ 27  │
├────┼────────┼─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
├────┼────────┼─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
│ 6  │ rho    │ 38  │
│ 7  │ omega  │ 86  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewComplex[LARGE]=[
┌───────────────────┐
│ complex table     │
├────┬────────┬─────┤
│ id │ name   │ age │
╞════╪════════╪═════╡
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
├────┼────────┼─────┤
│ 6  │ rho    │ 38  │
├────┼────────┼─────┤
│ 7  │ omega  │ 86  │
├────┼────────┼─────┤
│ 8  │ sigma  │ 64  │
├────┼────────┼─────┤
│ 9  │ lambda │ 12  │
├────┼────────┼─────┤
│ 10 │ beta   │ 27  │
╞════╪════════╪═════╡
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
╞════╪════════╪═════╡
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
├────┼────────┼─────┤
│ 6  │ rho    │ 38  │
├────┼────────┼─────┤
│ 7  │ omega  │ 86  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewComplex[MINIMAL]=[
complex table
id │ name   │ age
───┼────────┼────
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
6  │ rho    │ 38 
7  │ omega  │ 86 
8  │ sigma  │ 64 
9  │ lambda │ 12 
10 │ beta   │ 27 
───┼────────┼────
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
───┼────────┼────
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
6  │ rho    │ 38 
7  │ omega  │ 86 
]


tanoshi.utils.tables.TableViewTest.viewSimple[DEFAULT]=[
┌────┬────────┬─────┐
│ id │ name   │ age │
├────┼────────┼─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
│ 6  │ rho    │ 38  │
│ 7  │ omega  │ 86  │
│ 8  │ sigma  │ 64  │
│ 9  │ lambda │ 12  │
│ 10 │ beta   │ 27  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewSimple[LARGE]=[
┌────┬────────┬─────┐
│ id │ name   │ age │
╞════╪════════╪═════╡
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
├────┼────────┼─────┤
│ 6  │ rho    │ 38  │
├────┼────────┼─────┤
│ 7  │ omega  │ 86  │
├────┼────────┼─────┤
│ 8  │ sigma  │ 64  │
├────┼────────┼─────┤
│ 9  │ lambda │ 12  │
├────┼────────┼─────┤
│ 10 │ beta   │ 27  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewSimple[MINIMAL]=[
id │ name   │ age
───┼────────┼────
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
6  │ rho    │ 38 
7  │ omega  │ 86 
8  │ sigma  │ 64 
9  │ lambda │ 12 
10 │ beta   │ 27 
]


tanoshi.utils.tables.TableViewTest.viewTitleComplex[DEFAULT]=[
┌───────────────────┐
│ complex table     │
├────┬────────┬─────┤
│ id │ name   │ age │
├────┴────────┴─────┤
│ table 1           │
├────┬────────┬─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
│ 6  │ rho    │ 38  │
│ 7  │ omega  │ 86  │
│ 8  │ sigma  │ 64  │
│ 9  │ lambda │ 12  │
│ 10 │ beta   │ 27  │
├────┴────────┴─────┤
│ table 2           │
├────┬────────┬─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
├────┴────────┴─────┤
│ table 3           │
├────┬────────┬─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
│ 6  │ rho    │ 38  │
│ 7  │ omega  │ 86  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewTitleComplex[LARGE]=[
┌───────────────────┐
│ complex table     │
├────┬────────┬─────┤
│ id │ name   │ age │
╞════╧════════╧═════╡
│ table 1           │
├────┬────────┬─────┤
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
├────┼────────┼─────┤
│ 6  │ rho    │ 38  │
├────┼────────┼─────┤
│ 7  │ omega  │ 86  │
├────┼────────┼─────┤
│ 8  │ sigma  │ 64  │
├────┼────────┼─────┤
│ 9  │ lambda │ 12  │
├────┼────────┼─────┤
│ 10 │ beta   │ 27  │
╞════╧════════╧═════╡
│ table 2           │
├────┬────────┬─────┤
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
╞════╧════════╧═════╡
│ table 3           │
├────┬────────┬─────┤
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
├────┼────────┼─────┤
│ 6  │ rho    │ 38  │
├────┼────────┼─────┤
│ 7  │ omega  │ 86  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewTitleComplex[MINIMAL]=[
complex table
id │ name   │ age
───┴────────┴────
table 1
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
6  │ rho    │ 38 
7  │ omega  │ 86 
8  │ sigma  │ 64 
9  │ lambda │ 12 
10 │ beta   │ 27 
───┴────────┴────
table 2
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
───┴────────┴────
table 3
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
6  │ rho    │ 38 
7  │ omega  │ 86 
]


tanoshi.utils.tables.TableViewTest.viewTitleSimple[DEFAULT]=[
┌───────────────────┐
│ title             │
├────┬────────┬─────┤
│ id │ name   │ age │
├────┼────────┼─────┤
│ 1  │ phi    │ 33  │
│ 2  │ zeta   │ 20  │
│ 3  │ gamma  │ 93  │
│ 4  │ beta   │ 49  │
│ 5  │ kappa  │ 37  │
│ 6  │ rho    │ 38  │
│ 7  │ omega  │ 86  │
│ 8  │ sigma  │ 64  │
│ 9  │ lambda │ 12  │
│ 10 │ beta   │ 27  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewTitleSimple[LARGE]=[
┌───────────────────┐
│ title             │
├────┬────────┬─────┤
│ id │ name   │ age │
╞════╪════════╪═════╡
│ 1  │ phi    │ 33  │
├────┼────────┼─────┤
│ 2  │ zeta   │ 20  │
├────┼────────┼─────┤
│ 3  │ gamma  │ 93  │
├────┼────────┼─────┤
│ 4  │ beta   │ 49  │
├────┼────────┼─────┤
│ 5  │ kappa  │ 37  │
├────┼────────┼─────┤
│ 6  │ rho    │ 38  │
├────┼────────┼─────┤
│ 7  │ omega  │ 86  │
├────┼────────┼─────┤
│ 8  │ sigma  │ 64  │
├────┼────────┼─────┤
│ 9  │ lambda │ 12  │
├────┼────────┼─────┤
│ 10 │ beta   │ 27  │
└────┴────────┴─────┘
]


tanoshi.utils.tables.TableViewTest.viewTitleSimple[MINIMAL]=[
title
id │ name   │ age
───┼────────┼────
1  │ phi    │ 33 
2  │ zeta   │ 20 
3  │ gamma  │ 93 
4  │ beta   │ 49 
5  │ kappa  │ 37 
6  │ rho    │ 38 
7  │ omega  │ 86 
8  │ sigma  │ 64 
9  │ lambda │ 12 
10 │ beta   │ 27 
]